x-pack/plugins/lists/public/exceptions/hooks/use_exception_list.tsx

```diff
@@ -24,32 +32,50 @@ export type ReturnExceptionListAndItems = [boolean, ExceptionListAndItems | null
  */
 export const useExceptionList = ({
   http,
-  id,
-  namespaceType,
+  lists,
   pagination = {
     page: 1,
-    perPage: 20,
+    perPage: 5,
     total: 0,
   },
   filterOptions = {
     filter: '',
     tags: [],
   },
   onError,
+  dispatchListsInReducer,
 }: UseExceptionListProps): ReturnExceptionListAndItems => {
-  const [exceptionListAndItems, setExceptionList] = useState<ExceptionListAndItems | null>(null);
-  const [shouldRefresh, setRefresh] = useState<boolean>(true);
-  const refreshExceptionList = useCallback(() => setRefresh(true), [setRefresh]);
+  const [exceptionLists, setExceptionLists] = useState<
+    Array<ExceptionListSchema & { totalItems: number }>
+  >([]);
+  const [exceptionItems, setExceptionListItems] = useState<ExceptionListItemSchema[]>([]);
+  const [paginationInfo, setPagination] = useState<Pagination>(pagination);
+  const fetchExceptionList = useRef<Func | null>(null);
   const [loading, setLoading] = useState(true);
-  const tags = filterOptions.tags.sort().join();
+  const tags = useMemo(() => filterOptions.tags.sort().join(), [filterOptions.tags]);
+  const listIds = useMemo(
+    () =>
+      lists
+        .map((t) => t.id)
+        .sort()
+        .join(),
+    [lists]
+  );
 
   useEffect(
     () => {
-      let isSubscribed = true;
-      const abortCtrl = new AbortController();
+      let isSubscribed = false;
+      let abortCtrl: AbortController;
+
+      const fetchLists = async (): Promise<void> => {
+        isSubscribed = true;
+        abortCtrl = new AbortController();
 
-      const fetchData = async (idToFetch: string): Promise<void> => {
-        if (shouldRefresh) {
+        // TODO: workaround until api updated, will be cleaned up
+        let exceptions: ExceptionListItemSchema[] = [];
```

Comment:
There's a few icky things here. It's temp until the _find endpoint is updated to take an array of lists.